File: ce/outlook-app/snippets/csharp/CRMV8/query/cs/fetchaggregationandgroupby2.cs
// <snippetfetchaggregationandgroupby2>


 // *****************************************************************************************************************
 //                FetchXML      opportunity_count   Aggregate 2
 // *****************************************************************************************************************
 // Fetch the count of all opportunities.  This is the equivalent of
 // SELECT COUNT(*) AS opportunity_count ... in SQL.
 string opportunity_count = @" 
 <fetch distinct='false' mapping='logical' aggregate='true'> 
     <entity name='opportunity'> 
        <attribute name='name' alias='opportunity_count' aggregate='count'/> 
     </entity> 
 </fetch>";

 EntityCollection opportunity_count_result = _serviceProxy.RetrieveMultiple(new FetchExpression(opportunity_count));

 foreach (var c in opportunity_count_result.Entities)
 {
     Int32 aggregate2 = (Int32)((AliasedValue)c["opportunity_count"]).Value;
     System.Console.WriteLine("Count of all opportunities: " + aggregate2); 

 }
 System.Console.WriteLine("===============================");

// </snippetfetchaggregationandgroupby2>

File: ce/customer-engagement/on-premises/snippets/visualbasic/CRMV8/queryvb/vb/fetchaggregationandgroupby2.vb
' <snippetfetchaggregationandgroupby2>


 ' *****************************************************************************************************************
 '                FetchXML      opportunity_count   Aggregate 2
 ' *****************************************************************************************************************
 ' Fetch the count of all opportunities.  This is the equivalent of
 ' SELECT COUNT(*) AS opportunity_count ... in SQL.
 Dim opportunity_count As String = " " &amp; ControlChars.CrLf &amp; _
     "                    <fetch distinct='false' mapping='logical' aggregate='true'> " &amp; ControlChars.CrLf &amp; _
     "                        <entity name='opportunity'> " &amp; ControlChars.CrLf &amp; _
     "                           <attribute name='name' alias='opportunity_count' aggregate='count'/> " &amp; ControlChars.CrLf &amp; _
     "                        </entity> " &amp; ControlChars.CrLf &amp; _
     "                    </fetch>"

 Dim opportunity_count_result As EntityCollection = _serviceProxy.RetrieveMultiple(New FetchExpression(opportunity_count))

 For Each c In opportunity_count_result.Entities
     Dim aggregate2 As Int32 = CInt(Fix((CType(c("opportunity_count"), AliasedValue)).Value))
     Console.WriteLine("Count of all opportunities: " &amp; aggregate2)

 Next c
 Console.WriteLine("===============================")

' </snippetfetchaggregationandgroupby2>